examples/run/cuda/throughput_seq_example_cuda.cpp

```diff
@@ -0,0 +1,166 @@
+    vecmem::cuda::device_memory_resource device_mr;
+    traccc::memory_resource mr{device_mr, &host_mr};
+
+    vecmem::cuda::copy copy;
+
+    traccc::device::container_h2d_copy_alg<traccc::cell_container_types>
+        cell_h2d{mr, copy};
+    traccc::cuda::clusterization_algorithm ca(mr);
+    traccc::cuda::seeding_algorithm sa(mr);
+    traccc::cuda::track_params_estimation tp(mr);
+
+    traccc::demonstrator_input cells_event_vec;
+    cells_event_vec.reserve(i_cfg.loaded_events);
+
+    traccc::performance::timing_info elapsedTimes;
+
+    {
+
+        traccc::performance::timer t("File reading", elapsedTimes);
+
+        cells_event_vec = traccc::io::read(
+            i_cfg.loaded_events, i_cfg.input_directory, i_cfg.detector_file,
+            i_cfg.digitization_config_file, i_cfg.input_data_format, &host_mr);
+    }
+    // Seed random number generator
+    std::srand(std::time(0));
+
+    // Dummy count uses output of tp algorithm to ensure the compiler
+    // optimisations don't skip this step
+    std::size_t dummy_count = 0;
+
+    // Cold Run events
+    for (int i = 0; i < i_cfg.cold_run_events; ++i) {
+        const int event = std::rand() % i_cfg.loaded_events;
+        traccc::cell_container_types::host& cells_per_event =
+            cells_event_vec[event];
+        const traccc::cell_container_types::buffer cells_buffer =
+            cell_h2d(traccc::get_data(cells_per_event));
+        auto spacepoints_buffer = ca(cells_buffer);
+        auto seeds_buffer = sa(spacepoints_buffer);
+        auto params_buffer = tp(spacepoints_buffer, seeds_buffer);
+        traccc::bound_track_parameters_collection_types::host params_copy;
+        copy(params_buffer, params_copy);
+        dummy_count += params_copy.size();
+    }
+
+    // Loop over events
+    for (int i = 0; i < i_cfg.processed_events; ++i) {
+
+        const int event = std::rand() % i_cfg.loaded_events;
+
+        {
+
+            traccc::performance::timer t("Event processing", elapsedTimes);
+
+            traccc::cell_container_types::host& cells_per_event =
+                cells_event_vec[event];
+
+            // Copy the cell data to the device.
+            const traccc::cell_container_types::buffer cells_buffer =
+                cell_h2d(traccc::get_data(cells_per_event));
+
+            /*-------------------------------------------
+                Clusterization & Spacepoint formation
+            -------------------------------------------*/
+
+            auto spacepoints_buffer = ca(cells_buffer);
+
+            /*-----------------------
+              Seeding algorithm
+              -----------------------*/
+
+            auto seeds_buffer = sa(spacepoints_buffer);
+
+            /*----------------------------
+              Track params estimation
+              ----------------------------*/
+
+            auto params_buffer = tp(spacepoints_buffer, seeds_buffer);
+
+            traccc::bound_track_parameters_collection_types::host params_copy;
```

Comment:
It could be worth defining this container outside of the for-loop. Would reduce the number of re-allocations a bit.